### PR TITLE
WIP: Wrote tests for deleting user with FB/Google auth. Still need to… { Continued in #8796 }

### PR DESCRIPTION
**Pull request by pllearns**
WIP: Wrote two tests for FB and Google user deletion, still need to update object generators correctly, these tests need a little more work. 

[//]: # (Put Issue #7417  or URL here, if applicable. This will automatically close the issue if your PR is merged in)

### Changes
[//]: # Added two tests for Google and Facebook authorized user deletion. Still working on the matching component/methods to test against. 

[//]: # 0976d018-2433-410c-8c6e-8dbb42160978

----
UUID:

### CODE REVIEW

File: test/api/v3/integration/user/DELETE-user.test.js [hunk 1]
```diff
@@ -30,6 +30,20 @@ describe('DELETE /user', () => {
     });
   });
 
+  it('does not need a password if logged in through Facebook', async () => {
+    await user.auth.facebook.id.del('/user', {
```
Comment (khipkin) [inline test/api/v3/integration/user/DELETE-user.test.js:34]: > **`DELETE /user does not need a password if logged in through Facebook:`**
>     TypeError: Cannot read property 'del' of undefined
>      at _callee3$ (test/api/v3/integration/user/DELETE-user.test.js:34:11)

It looks like `user.auth.facebook.id` is undefined here.

File: test/api/v3/integration/user/DELETE-user.test.js [hunk 2]
```diff
@@ -30,6 +30,20 @@ describe('DELETE /user', () => {
...
+      user,
+    });
+    await expect(checkExistence('users', user._id)).to.eventually.eql(false);
+  });
+
+  it('does not need a password if logged in through Google', async () => {
+    await user.auth.google.id.del('/user', {
```
Comment (khipkin) [inline test/api/v3/integration/user/DELETE-user.test.js:41]: > **`DELETE /user does not need a password if logged in through Google:`**
>     TypeError: Cannot read property 'del' of undefined
>      at _callee4$ (test/api/v3/integration/user/DELETE-user.test.js:41:11)

It looks like `user.auth.google.id` is also undefined.

File: test/api/v3/integration/user/DELETE-user.test.js (hunk 1 above)
Comment (pllearns) [reply test/api/v3/integration/user/DELETE-user.test.js:34]: yep, I figured that, (hence the WIP) I'm still looking at how I should use the requester to get the right id. I was looking at something else when I made this test. I'll update. Thanks!

File: test/api/v3/integration/user/DELETE-user.test.js (hunk 2 above)
Comment (pllearns) [reply test/api/v3/integration/user/DELETE-user.test.js:41]: same comment as above. Thanks again for looking!

File: test/api/v3/integration/user/DELETE-user.test.js (hunk 1 above)
Comment (TheHollidayInn) [reply test/api/v3/integration/user/DELETE-user.test.js:34]: I think what you want to do here is just call `user.del('/user')`

And before that, you will want to generate a user with the auth.facebook.id. Which I think you said you are working on.

Comment (pllearns) [reply test/api/v3/integration/user/DELETE-user.test.js:34]: Thanks, we are trying that and the test comes back with this error:
`path `auth.facebook.id` was not saved, as it's a protected path.`

Here is the updated test code: 

```js

  it('does not need a password if logged in through Social Network to delete account', async () => {
    await user.update({
      'auth.facebook.id': 'some-fb-id',
      'auth.google.id': 'some-google-id',
      'auth.local': { ok: true },
    });

    await user.del('/user', {
    });

    await user.sync();
    await expect(checkExistence('users', user._id)).to.eventually.eql(false);
  });
```

Comment (TheHollidayInn) [reply test/api/v3/integration/user/DELETE-user.test.js:34]: Yea, you will have to alter the generator (which uses the Mongoose Model directly, I believe). You will not be able to update the id via the user update route.

Comment (pllearns) [reply test/api/v3/integration/user/DELETE-user.test.js:34]: Thanks for that. Sorry for the late reply to this. I will go ahead and work on the generator. I'll update soon.

**Comment by SabreCat** [comment]
This WIP hasn't seen any updates in some time. I'm going to close it for now; @pllearns, if you'd like to resume, you need only say so!

**Pull request closed without merging**